src/Features/ActivityPanels.php

```diff
@@ -60,13 +60,72 @@ public function add_user_data_fields( $user_data_fields ) {
 		);
 	}
 
+	/**
+	 * Determines if there are out of stock or low stock products.
+	 *
+	 * @return boolean
+	 */
+	public function has_low_stock_products() {
+		global $wpdb;
+
+		// Bail early if store does not manage stock, or Woo version < 3.6 needs lookup tables.
+		if (
+			'yes' !== get_option( 'woocommerce_manage_stock' ) ||
+			version_compare( get_option( 'woocommerce_db_version', null ), '3.6', '<' )
+		) {
+			return false;
+		}
+
+		$stock   = absint( max( get_option( 'woocommerce_notify_low_stock_amount' ), 1 ) );
+		$nostock = absint( max( get_option( 'woocommerce_notify_no_stock_amount' ), 0 ) );
+
+		$transient_name   = 'woocommerce_admin_low_stock_count';
+		$lowinstock_count = get_transient( $transient_name );
+
+		if ( false === $lowinstock_count ) {
+			$lowinstock_count = (int) $wpdb->get_var(
+				$wpdb->prepare(
+					"SELECT COUNT( product_id )
+					FROM {$wpdb->wc_product_meta_lookup} AS lookup
+					INNER JOIN {$wpdb->posts} as posts ON lookup.product_id = posts.ID
```

Comment:
I think we'll need another join on the product meta table since each product can have an individual low product stock threshold.

See https://github.com/woocommerce/woocommerce-admin/blob/5bed94bf9e3775ec58af2a9cc64a35d3fe767bfb/src/API/Reports/Stock/Stats/DataStore.php#L74 for prior art.